projects/packages/my-jetpack/_inc/hooks/use-analytics/index.js

```diff
@@ -0,0 +1,51 @@
+/* global myJetpackInitialState */
+/**
+ * External dependencies
+ */
+import { useEffect } from 'react';
+import jetpackAnalytics from '@automattic/jetpack-analytics';
+import { useConnection } from '@automattic/jetpack-connection';
+
+const useAnalytics = () => {
+	const { apiRoot, apiNonce } = myJetpackInitialState;
+
+	const { isUserConnected, userConnectionData } = useConnection( {
+		apiRoot,
+		apiNonce,
+	} );
+
+	const { login, ID } = userConnectionData.currentUser.wpcomUser;
+	/**
+	 * Initialize tracks with user data.
+	 * Should run when we have a connected user.
+	 */
+	useEffect( () => {
+		if ( isUserConnected ) {
+			jetpackAnalytics.initialize( ID, login );
+		}
+	} );
+	const {
+		clearedIdentity,
+		ga,
+		mc,
+		pageView,
+		purchase,
+		setGoogleAnalyticsEnabled,
+		setMcAnalyticsEnabled,
+		setProperties,
+		tracks,
+	} = jetpackAnalytics;
+	return {
+		clearedIdentity,
+		ga,
+		mc,
+		pageView,
+		purchase,
+		setGoogleAnalyticsEnabled,
+		setMcAnalyticsEnabled,
+		setProperties,
+		tracks,
+	};
```

Comment:
Any particular reason why you are not exporting the whole  `jetpackAnalytics` object?

```suggestion
	return jetpackAnalytics;
```

Please take it as a simple suggestion. Feel free to ignore it.

Reply:
There are internal methods in the original object that do not make sense to be exposed